crates/ra_ide/src/syntax_highlighting.rs

```diff
@@ -539,6 +541,25 @@ fn highlight_element(
                 _ => h,
             }
         }
+        T![&] => {
+            let ref_expr = element.parent().and_then(ast::RefExpr::cast)?;
+            let expr = ref_expr.expr()?;
+            let field_expr = match expr {
+                ast::Expr::FieldExpr(fe) => fe,
+                _ => return None,
+            };
+
+            let expr = field_expr.expr()?;
+            let ty = sema.type_of_expr(&expr)?;
+            if !ty.is_packed(db) {
```

Comment:
Another bit of semantics. I think there should be zero calls to `is_packed` in highlighting code. It's fine for highlighting to know "some fields accesss might be unsafe". It's not OK for highlighting to know *why* they are unsafe.

Reply:
Done